src/java.base/share/classes/java/io/ObjectStreamClass.java

```diff
@@ -830,7 +830,7 @@ void readNonProxy(ObjectInputStream in)
             char tcode = (char) in.readByte();
             String fname = in.readUTF();
             String signature = ((tcode == 'L') || (tcode == '[')) ?
-                in.readTypeString() : new String(new char[] { tcode });
+                in.readTypeString() : String.valueOf(tcode);
```

Comment:
Since the result of String.valueOf here will be equal to one of the primitive signatures strings ("I", "J", ...) (otherwise ObjectStreamField will throw an exception) it might make sense to turn this into a switch expression and simplify the whole thing. I can't tell how performance sensitive this piece of code is, though.

Reply:
Hi, I'll check and add one more PR if it makes sense